lz4_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.10)
+project(lz4_vendor)
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_vendor_package REQUIRED)
+
+ament_add_default_options()
+
+list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+find_package(lz4 QUIET)
+
+ament_vendor(lz4_vendor
+  SATISFIED ${lz4_FOUND}
+  VCS_URL https://github.com/lz4/lz4.git
+  VCS_VERSION v1.9.4
+  SOURCE_SUBDIR build/cmake
+  CMAKE_ARGS
+    -DBUILD_STATIC_LIBS:BOOL=OFF
+    -DBUILD_SHARED_LIBS:BOOL=${BUILD_SHARED_LIBS}
+    -DLZ4_BUILD_LEGACY_LZ4C:BOOL=OFF
+    -DLZ4_BUILD_CLI:BOOL=OFF
```

Comment:
@clalancette According to the https://github.com/lz4/lz4/blob/v1.9.4/build/cmake/CMakeLists.txt 
If using these two flags settled to `OFF`
```
    -DLZ4_BUILD_LEGACY_LZ4C:BOOL=OFF
    -DLZ4_BUILD_CLI:BOOL=OFF
```
CMake will use only files from the `\lib` subfolder to compile liblz4 library.
Therefore the library will contain only those part which is licensed under the `BSD license`. Perhaps we can remove mentions about GPLv2 license in this case. Thoughts?
Maybe add a comment for these two flags that we settling them OFF to ensure to use only the BSD licensed part?

Reply:
> Therefore the library will contain only those part which is licensed under the `BSD license`. Perhaps we can remove mentions about GPLv2 license in this case. Thoughts?

Ah, that's good to know, thanks.  Let's talk about it at the meeting tomorrow.

> Maybe add a comment for these two flags that we settling them OFF to ensure to use only the BSD licensed part?

Yes, I think this is a good idea regardless.

Reply:
@clalancette There is one caveat with it. 
As far as I understand if `liblz4` library will be found from the Ubuntu distro it will not be compiled and these flags will not be taken into account.  And Since `LZ4_BUILD_LEGACY_LZ4C` and `LZ4_BUILD_CLI` settled to `ON` by default the `liblz4` library from Ubuntu distros contain `GPLv2`  code.

Reply:
> As far as I understand if `liblz4` library will be found from the Ubuntu distro it will not be compiled and these flags will not be taken into account. And Since `LZ4_BUILD_LEGACY_LZ4C` and `LZ4_BUILD_CLI` settled to `ON` by default the `liblz4` library from Ubuntu distros contain `GPLv2` code.

I don't think this is going to be a problem.

I pulled down the orig.tar.gz and debian.tar.xz from https://packages.ubuntu.com/noble/liblz4-1, and then built them locally with `debuild -uc -us`.  When building, I see the following:

```
make[3]: Entering directory '/home/ubuntu/lz4_compile/lz4-1.9.4/lib'
compiling static library
x86_64-linux-gnu-gcc -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/home/ubuntu/lz4_compile/lz4-1.9.4=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/home/ubuntu/lz4_compile/lz4-1.9.4=/usr/src/lz4-1.9.4-1 -Wdate-time -D_FORTIFY_SOURCE=3 -DLZ4F_PUBLISH_STATIC_FUNCTIONS  -Wdate-time -D_FORTIFY_SOURCE=3 -DLZ4F_PUBLISH_STATIC_FUNCTIONS -DXXH_NAMESPACE=LZ4_   -c lz4.c lz4file.c lz4frame.c lz4hc.c xxhash.c
compiling dynamic library 1.9.4
x86_64-linux-gnu-gcc -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/home/ubuntu/lz4_compile/lz4-1.9.4=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/home/ubuntu/lz4_compile/lz4-1.9.4=/usr/src/lz4-1.9.4-1 -Wdate-time -D_FORTIFY_SOURCE=3 -DLZ4F_PUBLISH_STATIC_FUNCTIONS  -Wdate-time -D_FORTIFY_SOURCE=3 -DLZ4F_PUBLISH_STATIC_FUNCTIONS -DXXH_NAMESPACE=LZ4_  -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -Wl,-z,relro -shared lz4.c lz4file.c lz4frame.c lz4hc.c xxhash.c -fPIC -fvisibility=hidden -Wl,-soname=liblz4.so.1 -o liblz4.so.1.9.4
```

As far as I can tell, both the static and the shared library only use files from the lib directory.  It is true that the rest of the build goes on to build the programs and so on, but those will never be directly linked against by `rosbag2` or `mcap`, so we should be fine.